### ISSUE TITLE
`barrier=fence` renders too heavy at zoom level 16, 17

### ISSUE THREAD

**Issue by PeterLeGras** [gravitystorm/openstreetmap-carto#4428]
The line weight of `barrier=fence` is too heavy at zoom level 16. The stroke width is only 1px but it is black. I believe the line stroke colour should change to a lighter grey colour, otherwise it visually competes with building outlines. A similar change should also be considered at zoom level 17.

Example:
Notice how the following way is rendered at z=16;
https://www.openstreetmap.org/way/956286718
https://www.openstreetmap.org/#map=16/-33.7290/150.4565

**Comment by imagico** [comment]
I don't think this suggestion is going to work.  We have a large number of uses of simple thin lines in relatively neutral colors and a delicate balance between those to make them distinguishable.  Specifically the outlines we draw around landcover and road polygons need to be clearly differentiated from the barrier line signature.  See #3976.

**Comment by Adamant36** [comment]
What about just not rendering them at Z16/Z17? I know in California I've stopped mapping fences in any even semi-populated area because it adds to much clutter. It's also pretty unnecessary. Especially at Z16. I don't seeing the fences in my Z16 example helps anyone. Even in less dense areas not having fences render at z16 wouldn't be that much of a loss. 
![Screenshot 2021-07-14 205154](https://user-images.githubusercontent.com/30259065/125726615-d3d53c36-7a21-4de6-b268-2bb9661335a8.png)

**Comment by PeterLeGras** [comment]
For z16/17, I would support either not rendering them, or having the same stroke weight/colour as the outline of `landuse=residential` (considering fences physically perform the same function as the limits to a residential area). I would rather fences be indistinguishable from residential bounds than visually compete with `power=line` for example.

**Comment by Adamant36** [comment]
"or having the same stroke weight/colour as the outline of landuse=residential"

I had thought about that, but we aren't suppose to map residential landuse at the plot level and it might be confusing for people that are trying deal with mistakes, or really just in general, jf they are rendered the same.

**Comment by boothym** [comment]
Don't think it looks too bad at z16, for example here in Edinburgh: https://www.openstreetmap.org/#map=16/55.9515/-3.2427

Certainly wouldn't support removing them from z17.
![fences](https://user-images.githubusercontent.com/16965772/130818032-814aa609-8491-44c6-960f-94d62c12a5b5.JPG)

**Comment by kocio-pl** [comment]
> What about just not rendering them at Z16/Z17?

I think that would be going too far - look at the gates all around your example, they would be floating in the space or should be also rendered later.

So making both fences and gates less rendered lighter sounds good to me.

The other idea would be to make a subtag to them, so we know when they can be rendered later without making changes in the rest of the cases, That could be hard to get the pattern which ones are usually small and densely packed, but it worked with lane/street side parking rendering, so maybe that could work here also.

**Comment by Adamant36** [comment]
>  think that would be going too far - look at the gates all around your example, they would be floating in the space or should be also rendered later.

Gates don't render at z16 yet. So it wouldn't be a problem at that level. Your probably confusing them power poles. 


> Making both fences and gates less rendered lighter sounds good to me.

I had thought about that, but making barriers lighter was already tried and understandably rejected for curbs and I figured there would be the same problems for why it understandably didn't work in that situation. Maybe there wouldn't be though. 

> The other idea would be to make a subtag to them, so we know when they can be rendered later without making changes in the rest of the cases

That's an option, but it would probably take years to get off the ground as a tag and there's zero guarantee that it would. Let alone that there would be consensus here to render it once it does. My guess is that there wouldn't be. I think that creating a new tag should be default way to fix map clutter either. In the case of lane/street side parking rendering, it ultimately just made the broader problem of how prominently parking is rendered not get dealt with. Which I don't want to see happen here. As I don't think it's necessarily an urban versus rural issue. Making it one is way overly reductionist.

**Comment by Adamant36** [comment]
> I was rather thinking about a fence blocking access to some area (say, fence around part of forest or around a lake).

@matkoniecz If someone is planning on driving their car down some access road that might be blocked by a fence then really the solution to that should be the road having proper access tags so they don't assume it's accessible. In cases where there's a fence but it can still be accessed then the fence being rendered at z16 is superficial at best. If not wrong tagging. As there shouldn't be a case of a fence going across a highway without there being a gate. 

It's not like they couldn't find all that out at z17 or z18 though. Otherwise, the same argument could made be made for why fences should be rendered at z15, z14 and so on. There's nothing inherently special about z16 that makes fences have to start rendering at that level. It's just assumed there is because it's the status quo and not following the status quo is always by default going to be discounted as bad. Even if it actually is or not.

**Comment by jeisenbe** [comment]
The current barrier width is not actually 1 pixel, it is 0.4 pixels. That’s why it looks medium gray instead of dark gray (`#444`). This is about the narrowest we can go. An alternative would be to use medium gray instead of dark black. See https://github.com/gravitystorm/openstreetmap-carto/blob/6a4cd4b5c5b589af2306b1f507671e78be4dd3ec/style/landcover.mss#L862-L865

In the future if we change to use high-resolution tiles, it could be possible to use an even narrower line, which would get better results.

**Comment by owiecc** [comment]
Maybe it is a good time to revisit the fence situation. High res tiles are now available so we could try the narrower line width. 

PS. Currently fences and walls are not rendered but gates are. This leads to a weird [floating gate situation](https://www.openstreetmap.org/#map=19/57.045493/9.926390). 
<img width="709" height="474" alt="Image" src="https://github.com/user-attachments/assets/9766a495-732d-4873-8983-c6d75b3792f8" />

**Comment by matkoniecz** [comment]
> PS. Currently fences and walls are not rendered but gates are.

typically fences ARE rendered, and if they would not be that would be offtopic here and would be for a different issue

I bet that fences are either (1) not mapped (2) not mapped properly (entered as areas, not lines)

**Comment by matkoniecz** [comment]
> High res tiles are now available

are you sure?

**Comment by owiecc** [comment]
My bad, I was looking at the vector tiles. What is the correct repo for reporting issues with the vector tiles (MapTiler)?

**Comment by matkoniecz** [comment]
no good idea, I would check OpenStreetMap forums thread about vector tiles

**Comment by imagico** [comment]
This has nothing to do with OSM-Carto.  Information on the tile layers on osm.org is available on https://wiki.openstreetmap.org/wiki/Featured_tile_layers - but be advised that a lot of the information on that wiki page is factually **wrong**.  There is, unfortunately, no really good, independent source of information on the featured tile layers available.

**Comment by matkoniecz** [comment]
> but be advised that a lot of the information on that wiki page is factually wrong

do you have list somewhere, even partial, of what is wrong?

**Comment by imagico** [comment]
Made some comments on https://wiki.openstreetmap.org/wiki/Talk:Featured_tile_layers#Factually_wrong_and_misleading_info
